clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2428,6 +2430,50 @@ public void recordChecksumVerificationFailure() {
     hostLevelIngestionStats.recordChecksumVerificationFailure();
   }
 
+  /**
+   * Records metrics for the original size of full-assembled records (key + value) and RMD by utilizing the field
+   * {@link ChunkedValueManifest#size}.
+   * Also records the ratio of assembled record size to maximum allowed size, which is intended to be used to alert
+   * customers about how close they are to hitting the size limit.
+   * @param keyLen The size of the record's key
+   * @param valueBytes {@link Put#putValue} which is expected to be a serialized {@link ChunkedValueManifest}
+   * @param rmdBytes {@link Put#replicationMetadataPayload} which can be a serialized {@link ChunkedValueManifest} if
+   * RMD chunking was enabled or just the RMD payload otherwise
+   */
+  protected void recordAssembledRecordSize(int keyLen, ByteBuffer valueBytes, ByteBuffer rmdBytes, long currentTimeMs) {
+    try {
+      // TODO: Check with Jialin on this
+      // if (ByteUtils.getIntHeaderFromByteBuffer(valueBytes) != CHUNK_MANIFEST_SCHEMA_ID) {
+      // LOGGER.error("Unable to deserialize putValue into a ChunkedValueManifest due to incorrect integer header");
+      // return;
+      // }
+
+      byte[] valueByteArray = ByteUtils.extractByteArray(valueBytes);
+      ChunkedValueManifest valueManifest = manifestSerializer.deserialize(valueByteArray, CHUNK_MANIFEST_SCHEMA_ID);
+      int recordSize = keyLen + valueManifest.getSize();
+      hostLevelIngestionStats.recordAssembledRecordSize(recordSize, currentTimeMs);
+      recordAssembledRecordSizeRatio(calculateAssembledRecordSizeRatio(recordSize), currentTimeMs);
+
+      if (rmdBytes == null || rmdBytes.remaining() == 0) {
+        return;
+      }
+
+      int rmdSize = rmdBytes.remaining();
+      if (ByteUtils.getIntHeaderFromByteBuffer(rmdBytes) == CHUNK_MANIFEST_SCHEMA_ID) {
+        byte[] rmdByteArray = ByteUtils.extractByteArray(rmdBytes);
+        ChunkedValueManifest rmdManifest = manifestSerializer.deserialize(rmdByteArray, CHUNK_MANIFEST_SCHEMA_ID);
+        rmdSize = rmdManifest.getSize();
+      }
+      hostLevelIngestionStats.recordAssembledRmdSize(rmdSize, currentTimeMs);
```

Comment:
Looks like we are creating different metrics on value and RMD, will we enforce RMD size checking as well or it is just auxiliary information for oncall to check cluster/store status?

Reply:
It's the latter. This RMD size will just be a separate metric just for our information.